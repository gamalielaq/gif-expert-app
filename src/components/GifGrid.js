import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from "../hooks/useFetchGifs"
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { data, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>

            { loading && 'Cargando....'}

            <div className='card-grid'>
                {
                    data.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.protoType = {
    category: PropTypes.string.isRequired
}