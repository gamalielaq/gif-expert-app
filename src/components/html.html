<!DOCTYPE html>
<html lang="es">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <div class="container"> 
        <h1>Hellow world</h1>
        <div class="main">
            Esto es el contenido de mi Web
        </div>
    </div>
    <footer>
        <div class="footer">
                <h5>Gamaliel Humberto Abanto Quiroz</h5>
        </div>
    </footer>
</body>
</html>