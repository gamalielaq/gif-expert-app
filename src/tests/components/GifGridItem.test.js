import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en GifGridItem', () => {

    const title = 'Un Titulo';
    const url = 'https://localhost/algo.jpg';
    const wraper = shallow( <GifGridItem title={title} url={url} /> );

    test('deve de mostrar el componente correctamene', () => {
        expect( wraper ).toMatchSnapshot();
    })

    test('deve de tener un pÃ¡rrarfo con el titulo', () => {
        const parrafo = wraper.find('p');
        expect( parrafo.text().trim()).toBe(title);
    })

    test('deve de tener una imaagen igual al url y alt de los props', () => {
        const image = wraper.find('img');
        expect(image.prop('src')).toBe(url);
        expect(image.prop('alt')).toBe(title);
    })

    test('deve tener la clase animate__fadeIn', () => {
        const div = wraper.find('div');
        const clasname = div.prop('className');
        expect(clasname.includes('animate__fadeIn')).toBe(true);
    })
});