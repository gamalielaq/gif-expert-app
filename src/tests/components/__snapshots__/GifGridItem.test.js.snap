// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem deve de mostrar el componente correctamene 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un Titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un Titulo
  </p>
</div>
`;
